Modules/signalmodule.c

```diff
@@ -1365,77 +1365,60 @@ ITIMER_PROF -- decrements both when the process is executing and\n\
 A signal handler function is called with two arguments:\n\
 the first is the signal number, the second is the interrupted stack frame.");
 
-static struct PyModuleDef signalmodule = {
-    PyModuleDef_HEAD_INIT,
-    "_signal",
-    module_doc,
-    -1,
-    signal_methods,
-    NULL,
-    NULL,
-    NULL,
-    NULL
-};
-
-PyMODINIT_FUNC
-PyInit__signal(void)
-{
-    PyObject *m, *d;
-    int i;
 
-    /* Create the module and add the functions */
-    m = PyModule_Create(&signalmodule);
-    if (m == NULL)
-        return NULL;
 
+static int
+signal_exec(PyObject *m)
+{
+    /* add the functions */
 #if defined(HAVE_SIGWAITINFO) || defined(HAVE_SIGTIMEDWAIT)
     if (!initialized) {
         if (PyStructSequence_InitType2(&SiginfoType, &struct_siginfo_desc) < 0)
```

Comment:
I would be nice to convert the static type SiginfoType to a heap type, but it can be done later.

Reply:
@vstinner what is the procedure for types defined using PyStructSequence_InitType2?

Reply:
I don't know.